test/integration-serial/serial_deploy_test.rb

```diff
@@ -235,6 +235,109 @@ def test_all_expected_statsd_metrics_emitted_with_essential_tags
     end
   end
 
+  def test_cr_deploys_without_rollout_config_when_none_present
+    skip if kube_server_version < Gem::Version.new('1.11.0')
+    assert_deploy_success(deploy_fixtures("crd", subset: %w(widgets.yml)))
+    assert_deploy_success(deploy_fixtures("crd", subset: %w(widgets_cr.yml)))
+    assert_logs_match_all([
+      "Don't know how to monitor resources of type Widget. Assuming Widget/my-first-widget deployed successfully.",
+    ])
+  ensure
+    wait_for_all_crd_deletion
+  end
+
+  def test_cr_success_with_default_rollout_config
+    skip if kube_server_version < Gem::Version.new('1.11.0')
+    assert_deploy_success(deploy_fixtures("crd", subset: ["with_default_params.yml"]))
+    success_patch = {
+      "status": {
+        "observedGeneration": 1,
+        "conditions": [
+          {
+            "type": "Ready",
+            "reason": "test",
+            "message": "test",
+            "status": "True",
+          },
+        ],
+      },
+    }.to_json
+
+    thread = Thread.new { get_and_patch_cr(kind: "parameterized", name: "with-default-params", patch: success_patch) }
```

Comment:
What happens if you don't do this in a separate thread?

Reply:
It turns out `deploy_fixtures` is more powerful than I thought at first. I've removed the crappy threaded code and used the optional block capability of `deploy_fixtures` instead :+1: